autocodesign/utils_test.go

```diff
@@ -0,0 +1,181 @@
+					},
+					UITestTargetProfilesByBundleID: nil,
+					Certificate:                    certificate,
+				},
+			},
+			expected: map[DistributionType]AppCodesignAssets{
+				Development: {
+					ArchivableTargetProfilesByBundleID: map[string]Profile{
+						"dev-1": profile("base", "1"),
+						"dev-2": profile("addition", "4"),
+					},
+					UITestTargetProfilesByBundleID: map[string]Profile{
+						"dev-uitest-1": profile("base", "2"),
+						"dev-uitest-2": profile("addition-uitest", "5"),
+					},
+					Certificate: certificate,
+				},
+				AppStore: {
+					ArchivableTargetProfilesByBundleID: map[string]Profile{
+						"app-store-1": profile("base", "3"),
+					},
+					UITestTargetProfilesByBundleID: nil,
+					Certificate:                    certificate,
+				},
+			},
+		},
+		{
+			name: "Base value is empty",
+			base: map[DistributionType]AppCodesignAssets{},
+			addition: map[DistributionType]AppCodesignAssets{
+				Enterprise: {
+					ArchivableTargetProfilesByBundleID: map[string]Profile{
+						"enterprise-1": profile("enterprise", "1"),
+					},
+					UITestTargetProfilesByBundleID: nil,
+					Certificate:                    certificate,
+				},
+			},
+			expected: map[DistributionType]AppCodesignAssets{
+				Enterprise: {
+					ArchivableTargetProfilesByBundleID: map[string]Profile{
+						"enterprise-1": profile("enterprise", "1"),
+					},
+					UITestTargetProfilesByBundleID: nil,
+					Certificate:                    certificate,
+				},
+			},
+		},
+		{
+			name: "Additional value is empty",
+			base: map[DistributionType]AppCodesignAssets{
+				AdHoc: {
+					ArchivableTargetProfilesByBundleID: map[string]Profile{
+						"ad-hoc-1": profile("ad-hoc", "1"),
+					},
+					UITestTargetProfilesByBundleID: nil,
+					Certificate:                    certificate,
+				},
+			},
+			addition: map[DistributionType]AppCodesignAssets{},
+			expected: map[DistributionType]AppCodesignAssets{
+				AdHoc: {
+					ArchivableTargetProfilesByBundleID: map[string]Profile{
+						"ad-hoc-1": profile("ad-hoc", "1"),
+					},
+					UITestTargetProfilesByBundleID: nil,
+					Certificate:                    certificate,
+				},
+			},
+		},
+		{
+			name:     "Empty values",
+			base:     map[DistributionType]AppCodesignAssets{},
+			addition: map[DistributionType]AppCodesignAssets{},
+			expected: map[DistributionType]AppCodesignAssets{},
+		},
+	}
+
+	for _, test := range tests {
+		merged := mergeCodeSignAssets(test.base, test.addition)
+		assert.Equal(t, test.expected, merged)
+	}
+}
+
+func profile(name, id string) testProfile {
+	return testProfile{
+		attributes: appstoreconnect.ProfileAttributes{
+			Name:           name,
+			UUID:           id,
+			ProfileContent: []byte{},
+			Platform:       "",
+			ExpirationDate: appstoreconnect.Time{},
+		},
+		id:             id,
+		bundleID:       fmt.Sprintf("bundle-id-%s", id),
+		certificateIDs: nil,
+		deviceIDs:      nil,
+	}
+}
+
+type testProfile struct {
```

Comment:
Could use newMockProfile() (https://github.com/bitrise-io/go-xcode/blob/1ca7ff962ac144cb6ebeb791fb9451cb7b148c58/autocodesign/autocodesign_test.go#L48)

Reply:
Updated in: https://github.com/bitrise-io/go-xcode/pull/135/commits/6c8fd2975b7dfcc30fb56db4ebcad3dd58ef54da